cpp/src/sort/sort.cu

```diff
@@ -72,6 +73,18 @@ std::unique_ptr<table> sort(table_view input,
                             rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
+
+  if (input.num_columns() == 1 && input.column(0).type().id() == type_id::STRING) {
+    auto col_order  = column_order.empty() ? order::ASCENDING : column_order.front();
+    auto precedence = null_precedence.empty() ? null_order::BEFORE : null_precedence.front();
+    auto input_col  = strings_column_view(input.column(0));
+    auto result =
+      cudf::strings::detail::sort(input_col, col_order, precedence, rmm::cuda_stream_default, mr);
+    std::vector<std::unique_ptr<column>> cols;
+    cols.emplace_back(std::move(result));
+    return std::make_unique<table>(std::move(cols));
+  }
```

Comment:
I think the `cudf::strings::detail::sort` should be made into `cudf::string::detail::sorted_order` and then the specialization logic should be put into `cudf::sorted_order` so that more places in the code can benefit from this optimization. After all, `sort` just calls `sorted_order` + `gather`.